pkgs/by-name/do/dorion/package.nix

```
@@ -1,65 +1,194 @@
-{ lib
-, stdenv
-, fetchurl
-, autoPatchelfHook
-, dpkg
-, glib-networking
-, gst_all_1
-, libappindicator
-, libayatana-appindicator
-, webkitgtk_4_0
-, wrapGAppsHook3
+{
+  lib,
+  fetchFromGitHub,
+  fetchurl,
+  rustPlatform,
+  cmake,
+  ninja,
+  wrapGAppsHook4,
+  glib-networking,
+  gst_all_1,
+  libsysprof-capture,
+  libayatana-appindicator,
+  nodejs,
+  openssl,
+  pkg-config,
+  yq-go,
+  pnpm_9,
+  webkitgtk_4_1,
+  cargo-tauri,
+  desktop-file-utils,
 }:
 
-stdenv.mkDerivation (finalAttrs: {
+let
+  webkitgtk_4_1' = webkitgtk_4_1.override {
+    enableExperimental = true;
+  };
+
+  shelter = fetchurl {
+    url = "https://raw.githubusercontent.com/uwu/shelter-builds/de7151650c240fc226d41ff898547edc4d23c94d/shelter.js";
+    hash = "sha256-dqAbq4O61t91gnVBOiN5KgJG3Frg9ePMPKjn9QWtkws=";
+    meta = {
+      homepage = "https://github.com/uwu/shelter";
+      sourceProvenance = [ lib.sourceTypes.binaryBytecode ]; # actually, minified JS
+      license = lib.licenses.cc0;
+    };
+  };
+in
+
+# nyo finalAttrs :<
+# https://github.com/NixOS/nixpkgs/pull/194475
+rustPlatform.buildRustPackage rec {
   pname = "dorion";
-  version = "5.0.1";
+  version = "6.4.1";
 
-  src = fetchurl {
-    url = "https://github.com/SpikeHD/Dorion/releases/download/v${finalAttrs.version }/Dorion_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-cCZikTZ+IU3mq/FkJfeggXLyWIsWG+a2qu1GbgW93WQ=";
+  src = fetchFromGitHub {
+    owner = "SpikeHD";
+    repo = "Dorion";
+    tag = "v${version}";
+    hash = "sha256-La58qj9acZzTG7WDfnBNdVWzAjOYVkIZzO8uyjQvt6o=";
   };
 
-  unpackCmd = ''
-    dpkg -X $curSrc .
-  '';
+  cargoPatches = [
+    ./no-cargo-patch.patch
+  ];
 
-  runtimeDependencies = [
-    glib-networking
-    libappindicator
-    libayatana-appindicator
+  cargoRoot = "src-tauri";
+  buildAndTestSubdir = cargoRoot;
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-5qK1IRaUrf4pYDmoBZoR6q6rL06irurwo4v5JLX+Io4=";
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit pname version src;
+    hash = "sha256-xBonUzA4+1zbViEsKap6CaG6ZRldW1LjNYIB+FmVRFs=";
+  };
+
+  # CMake (webkit extension)
+  cmakeDir = ".";
+  cmakeBuildDir = "src-tauri/extension_webkit";
+  dontUseCmakeConfigure = true;
+  dontUseNinjaBuild = true;
+  dontUseNinjaCheck = true;
+  dontUseNinjaInstall = true;
+  # cmake's supposed to set this automatically
+  # ... but the detection is based on the presence of ninja build hook
+  cmakeFlags = [
+    "-GNinja"
   ];
 
   nativeBuildInputs = [
-    autoPatchelfHook
-    dpkg
-    wrapGAppsHook3
+    pnpm_9.configHook
+    cargo-tauri.hook
+    nodejs
+    pkg-config
+    wrapGAppsHook4
+    yq-go
+    desktop-file-utils
+    cmake
+    ninja
   ];
 
   buildInputs = [
-    glib-networking
+    openssl
+    webkitgtk_4_1'
     gst_all_1.gst-plugins-bad
     gst_all_1.gst-plugins-base
     gst_all_1.gst-plugins-good
```

```suggestion
    gst_all_1.gst-plugins-good
    gst_all_1.gst-plugins-rs
```
The log also reports `GStreamer element audiornnoise not found. Please install it`. This should provide it

---

So remove gst-plugins-bad correct?

---

> So remove gst-plugins-bad correct?

Nope, we definitely need them for webp, webrtc, etc, so what is your suggestion?

---

Add `gst_all_1.gst-plugins-rs`

---

Done